apis/r/R/SOMASparseNDArray.R

```diff
@@ -74,7 +78,76 @@ SOMASparseNDArray <- R6::R6Class(
         j = bit64::as.integer64(values@j),
         x = values@x
       )
-      colnames(coo) <- c(self$dimnames(), self$attrnames())
+      dnames <- self$dimnames()
+      colnames(coo) <- c(dnames, self$attrnames())
+      ranges <- sapply(
+        X = dnames,
+        FUN = function(x) {
+          return(range(coo[[x]]))
+        },
+        simplify = FALSE,
+        USE.NAMES = TRUE
+      )
+      bbox <- bbox %||% ranges
+      if (is.null(names(bbox))) {
+        names(bbox) <- dnames
+      }
+      if (!is_named(bbox, allow_empty = FALSE)) {
+        idx <- which(!nzchar(names(bbox)))
+        names(bbox)[idx] <- dnames[idx]
+      }
+      bbox <- bbox[dnames]
+      if (length(bbox) != length(dnames)) {
+        stop("dnames")
```

Comment:
can be more informative to the person who will read the error message

Reply:
Added in 9b3ea56